GetUrl.cs

```diff
@@ -134,29 +127,29 @@ private async Task<string> GetUrlFromBackend(string bcchCommand)
                 {
                     err = err.Replace("#< CLIXML\n", "");
                     var deserialized = PSSerializer.DeserializeAsList(err);
-                    err = String.Join("", deserialized);
-                    err = Regex.Replace(err, "\\u001b\\[[0-9]{1,3}m", "");
+                    err = string.Join("", deserialized);
+                    err = FixedRegex().Replace(err, "");
                     throw new Exception(err);
                 }
             }
             return await process.StandardOutput.ReadToEndAsync();
         }
 
-        private string validateTextParam(string paramName, string paramValue)
+        private static string ValidateTextParam(string paramName, string paramValue)
         {
             var validatedParam = "";
             if (!string.IsNullOrEmpty(paramValue))
                 validatedParam = $" -{paramName} \"{paramValue}\"";
             return validatedParam;

```

Comment:
```suggestion
            return !string.IsNullOrEmpty(paramValue) ? $" -{paramName} \"{paramValue}\"" : string.Empty;
```

We could maybe further improve this function with eliminating the `validatedParam` property combined with a ternary operator. This could be up for debate and maybe depend on the personal preference of using a ternary operator.